test/opt/test_fold_spec_const_op_composite.cpp

```diff
@@ -0,0 +1,922 @@
+
+                "%unsigned_31 = OpConstant %uint 31",
+                "%v2unsigned_31 = OpConstantComposite %v2uint %unsigned_31 %unsigned_31",
+                "%unsigned_left_shift_max = OpSpecConstantOp %v2uint ShiftLeftLogical %unsigned_one_vec %v2unsigned_31",
+                "%unsigned_right_shift_logical = OpSpecConstantOp %v2uint ShiftRightLogical %unsigned_left_shift_max %v2unsigned_31",
+                "%signed_right_shift_arithmetic = OpSpecConstantOp %v2int ShiftRightArithmetic %unsigned_left_shift_max %v2unsigned_31",
+              },
+              // expected
+              {
+                "%58 = OpConstant %int 2",
+                "%59 = OpConstant %int 2",
+                "%xor_1_3 = OpConstantComposite %v2int %58 %59",
+                "%61 = OpConstant %int 0",
+                "%62 = OpConstant %int 0",
+                "%and_1_2 = OpConstantComposite %v2int %61 %62",
+                "%64 = OpConstant %int 3",
+                "%65 = OpConstant %int 3",
+                "%or_1_2 = OpConstantComposite %v2int %64 %65",
+
+                "%unsigned_31 = OpConstant %uint 31",
+                "%v2unsigned_31 = OpConstantComposite %v2uint %unsigned_31 %unsigned_31",
+                "%67 = OpConstant %uint 2147483648",
+                "%68 = OpConstant %uint 2147483648",
+                "%unsigned_left_shift_max = OpConstantComposite %v2uint %67 %68",
+                "%70 = OpConstant %uint 1",
+                "%71 = OpConstant %uint 1",
+                "%unsigned_right_shift_logical = OpConstantComposite %v2uint %70 %71",
+                "%73 = OpConstant %int -1",
+                "%74 = OpConstant %int -1",
+                "%signed_right_shift_arithmetic = OpConstantComposite %v2int %73 %74",
+              },
+            },
+        // clang-format on
+    })));
+
+// Tests the swizzle operations for spec const vectors.
+INSTANTIATE_TEST_CASE_P(
+    Swizzle, FoldSpecConstantOpAndCompositePassTest,
+    ::testing::ValuesIn(std::vector<
+                        FoldSpecConstantOpAndCompositePassTestCase>({
+        // clang-format off
+            // Get scalar
+            {
+              // original
+              {
+                "%x = OpSpecConstantOp %int CompositeExtract %v4int_0_1_2_3 0",
+                "%y = OpSpecConstantOp %int CompositeExtract %v4int_0_1_2_3 1",
+                "%z = OpSpecConstantOp %int CompositeExtract %v4int_0_1_2_3 2",
+                "%w = OpSpecConstantOp %int CompositeExtract %v4int_0_1_2_3 3",
+              },
+              // expected
+              {
+                "%x = OpConstant %int 0",
+                "%y = OpConstant %int 1",
+                "%z = OpConstant %int 2",
+                "%w = OpConstant %int 3",
+              },
+            },
+            // Get vector
+            {
+              // original
+              {
+                "%xy = OpSpecConstantOp %v2int VectorShuffle %v4int_0_1_2_3 %v4int_0_1_2_3 0 1",
```

Comment:
All the cases are extracting from vector of 4 elements. What about a case extracting from 2 one-element vectors to form a four-element vector.